lib/perl/Genome/InstrumentData/AlignmentResult/Merged/Speedseq.pm

```diff
@@ -0,0 +1,129 @@
+            is_optional=>1,
+        },
+        aligner_params => {
+            is => 'Text',
+            is_optional=>1,
+            doc => 'any additional params for the aligner in a single string',
+        },
+    ],
+    has_param => [
+        samtools_version => {
+            is=>'Text',
+            is_optional=>1,
+            doc=>'Version of samtools to use when creating BAM files',
+        },
+        picard_version => {
+            is=>'Text',
+            is_optional=>1,
+            doc=>'Version of picard to use when creating bam files',
+        },
+    ],
+    has_calculated => [
+        _final_bam_file => {
+            is => 'Text', calculate_from => ['temp_staging_directory', 'id',],
+            calculate => q{ return join('/', $temp_staging_directory, $id . '.bam'); },
+        },
+        merged_alignment_bam_path => {
+            is => 'Text', calculate_from => ['output_dir', 'id'],
+            calculate => q{ return join('/', $output_dir, $id . '.bam'); }
+        },
+    ]
+};
+
+sub create {
+    my $class = shift;
+
+    my $self = $class->SUPER::create(@_);
+    return unless ($self);
+
+    $self->_prepare_staging_directory;
+    my $temp_directory = Genome::Sys->create_temp_directory();
+
+    my $aligner_index = Genome::InstrumentData::AlignmentResult::get_reference_sequence_index($self);
+
+    my %params = (
+        bams => [ map {$_->bam_path} $self->instrument_data ],
+        reference_fasta => $aligner_index->full_consensus_path('fa'),
+        output_prefix => File::Spec->join($self->temp_staging_directory, $self->id),
+        version => $self->aligner_version,
+        sort_memory => $self->sort_memory,
+        temp_directory => $temp_directory,
+    );
+    my $command = Genome::Model::Tools::Speedseq::Realign->create(%params);
+    unless ($command->execute) {
+        die $self->error_message('Failed to execute Speedseq realign for instrument data ' . $self->instrument_data->id);
+    }
+
+    my $final_bam = $self->_final_bam_file,
+    $self->debug_message("Indexing the final BAM file...");
+    my $index_cmd = Genome::Model::Tools::Sam::IndexBam->create(
+        bam_file    => $final_bam,
+        use_version => $self->samtools_version,
+    );
+    my $index_cmd_rv = $index_cmd->execute;
+
+    if($index_cmd_rv ne 1) {
+        #not failing here because this is not a critical error.  this can be regenerated manually if needed.
+        $self->warning_message('Failed to create bam index for ' . $final_bam);
+    }
+
+    $self->create_bam_md5($final_bam);
+
+    $self->_prepare_output_directory;
+    $self->_promote_data;
+    $self->_reallocate_disk_allocation;
+
+    return $self;
+}
+
+sub sort_memory {
+    my $self = shift;
+
+    #This needs to match the LSF resource request or during testing needs to
+    #be 3 or so
+    #We might want to set an lsf resource to be at least 20G which is the
+    #default for speedseq -M parameter
+    return '3';
```

Comment:
Should this be configurable somehow?  The comment suggests so.  (If not, is `3` appropriate in production?)